apps/costcontrol/js/settings/autosettings.js

```diff
@@ -142,14 +142,21 @@ var AutoSettings = (function() {
         }
       );
 
-      // Add an event listener to switch the option
-      guiWidget.addEventListener('change', function _onContentChange() {
-        var value = guiWidget.value;
-        if (guiWidget.type === 'number') {
-          value = parseFloat(value);
-        }
-        settings.option(optionName, value);
-      });
+      if (guiWidget.type !== 'date') {
```

Comment:
I think you could avoid some code duplication if all depends on `guiWidget.type`. The listener is almost the same for both events and the event itself only depends of the type of the widget.